### DISCUSSION TITLE
HTTP2 Protocol error

### DISCUSSION THREAD

**Discussion by marcoroganovic** [chimurai/http-proxy-middleware#571] [Q&A]
### Is this a question?

Yes.

I'm getting protocol error on some pages after transforming them with `http-proxy-response-rewrite` package, has anyone encountered a similar issue? Can the payload side present the problem, it has more than 350000 characters.

**Comment by marcoroganovic** [comment]
I have found the solution, the package was not removing the content-length header from the proxy response object while modifying it.

**Comment by ganmor** [comment]
Could you elaborate on that ?
we have http2 error in chrome when proxing to aws cloudfront.
This is only happening after the server has been running for while.

**Comment by marcoroganovic** [comment]
Basically, I've just added this line of code in `onProxyRes` callback:

```javascript
delete proxyRes.headers['content-length'];
```

Not sure if we faced same error though.
